beacon-chain/forkchoice/doubly-linked-tree/forkchoice.go

```diff
@@ -172,8 +171,13 @@ func (f *ForkChoice) updateCheckpoints(ctx context.Context, jc, fc *ethpb.Checkp
 			currentSlot := slots.CurrentSlot(f.store.genesisTime)
 			if slots.SinceEpochStarts(currentSlot) < params.BeaconConfig().SafeSlotsToUpdateJustified {
 				f.store.prevJustifiedCheckpoint = f.store.justifiedCheckpoint
+				root := bytesutil.ToBytes32(jc.Root)
 				f.store.justifiedCheckpoint = &forkchoicetypes.Checkpoint{Epoch: jc.Epoch,
-					Root: bytesutil.ToBytes32(jc.Root)}
+					Root: root}
+				if err := f.updateJustifiedBalances(ctx, root); err != nil {
+					f.store.checkpointsLock.Unlock()
+					return errors.Wrap(err, "could not update justified balances")
```

Comment:
This error message appears at least three times (not counting tests). This is my threshold of moving it to a package-level constant.